// fastly/dist/model/AclEntryResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AclEntry = _interopRequireDefault(require("./AclEntry"));
var _AclEntryResponseAllOf = _interopRequireDefault(require("./AclEntryResponseAllOf"));
var _Timestamps = _interopRequireDefault(require("./Timestamps"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The AclEntryResponse model module.
 * @module model/AclEntryResponse
 * @version 4.1.1
 */
var AclEntryResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AclEntryResponse</code>.
   * @alias module:model/AclEntryResponse
   * @implements module:model/AclEntry
   * @implements module:model/Timestamps
   * @implements module:model/AclEntryResponseAllOf
   */
  function AclEntryResponse() {
    _classCallCheck(this, AclEntryResponse);
    _AclEntry["default"].initialize(this);
    _Timestamps["default"].initialize(this);
    _AclEntryResponseAllOf["default"].initialize(this);
    AclEntryResponse.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(AclEntryResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>AclEntryResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AclEntryResponse} obj Optional instance to populate.
     * @return {module:model/AclEntryResponse} The populated <code>AclEntryResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AclEntryResponse();
        _AclEntry["default"].constructFromObject(data, obj);
        _Timestamps["default"].constructFromObject(data, obj);
        _AclEntryResponseAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('negated')) {
          obj['negated'] = _ApiClient["default"].convertToType(data['negated'], 'Number');
        }
        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }
        if (data.hasOwnProperty('ip')) {
          obj['ip'] = _ApiClient["default"].convertToType(data['ip'], 'String');
        }
        if (data.hasOwnProperty('subnet')) {
          obj['subnet'] = _ApiClient["default"].convertToType(data['subnet'], 'Number');
        }
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('deleted_at')) {
          obj['deleted_at'] = _ApiClient["default"].convertToType(data['deleted_at'], 'Date');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
        if (data.hasOwnProperty('acl_id')) {
          obj['acl_id'] = _ApiClient["default"].convertToType(data['acl_id'], 'String');
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('service_id')) {
          obj['service_id'] = _ApiClient["default"].convertToType(data['service_id'], 'String');
        }
      }
      return obj;
    }
  }]);
  return AclEntryResponse;
}();
/**
 * Whether to negate the match. Useful primarily when creating individual exceptions to larger subnets.
 * @member {module:model/AclEntryResponse.NegatedEnum} negated
 * @default NegatedEnum.0
 */
AclEntryResponse.prototype['negated'] = undefined;

/**
 * A freeform descriptive note.
 * @member {String} comment
 */
AclEntryResponse.prototype['comment'] = undefined;

/**
 * An IP address.
 * @member {String} ip
 */
AclEntryResponse.prototype['ip'] = undefined;

/**
 * Number of bits for the subnet mask applied to the IP address. For IPv4 addresses, a value of 32 represents the smallest subnet mask (1 address), 24 represents a class C subnet mask (256 addresses), 16 represents a class B subnet mask (65k addresses), and 8 is class A subnet mask (16m addresses). If not provided, no mask is applied.
 * @member {Number} subnet
 */
AclEntryResponse.prototype['subnet'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
AclEntryResponse.prototype['created_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */
AclEntryResponse.prototype['deleted_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
AclEntryResponse.prototype['updated_at'] = undefined;

/**
 * @member {String} acl_id
 */
AclEntryResponse.prototype['acl_id'] = undefined;

/**
 * @member {String} id
 */
AclEntryResponse.prototype['id'] = undefined;

/**
 * @member {String} service_id
 */
AclEntryResponse.prototype['service_id'] = undefined;

// Implement AclEntry interface:
/**
 * Whether to negate the match. Useful primarily when creating individual exceptions to larger subnets.
 * @member {module:model/AclEntry.NegatedEnum} negated
 * @default NegatedEnum.0
 */
_AclEntry["default"].prototype['negated'] = undefined;
/**
 * A freeform descriptive note.
 * @member {String} comment
 */
_AclEntry["default"].prototype['comment'] = undefined;
/**
 * An IP address.
 * @member {String} ip
 */
_AclEntry["default"].prototype['ip'] = undefined;
/**
 * Number of bits for the subnet mask applied to the IP address. For IPv4 addresses, a value of 32 represents the smallest subnet mask (1 address), 24 represents a class C subnet mask (256 addresses), 16 represents a class B subnet mask (65k addresses), and 8 is class A subnet mask (16m addresses). If not provided, no mask is applied.
 * @member {Number} subnet
 */
_AclEntry["default"].prototype['subnet'] = undefined;
// Implement Timestamps interface:
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
_Timestamps["default"].prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */
_Timestamps["default"].prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
_Timestamps["default"].prototype['updated_at'] = undefined;
// Implement AclEntryResponseAllOf interface:
/**
 * @member {String} acl_id
 */
_AclEntryResponseAllOf["default"].prototype['acl_id'] = undefined;
/**
 * @member {String} id
 */
_AclEntryResponseAllOf["default"].prototype['id'] = undefined;
/**
 * @member {String} service_id
 */
_AclEntryResponseAllOf["default"].prototype['service_id'] = undefined;

/**
 * Allowed values for the <code>negated</code> property.
 * @enum {Number}
 * @readonly
 */
AclEntryResponse['NegatedEnum'] = {
  /**
   * value: 0
   * @const
   */
  "0": 0,
  /**
   * value: 1
   * @const
   */
  "1": 1
};
var _default = AclEntryResponse;
exports["default"] = _default;
