// fastly/dist/model/ApexRedirect.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ApexRedirectAllOf = _interopRequireDefault(require("./ApexRedirectAllOf"));
var _ServiceIdAndVersion = _interopRequireDefault(require("./ServiceIdAndVersion"));
var _Timestamps = _interopRequireDefault(require("./Timestamps"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The ApexRedirect model module.
 * @module model/ApexRedirect
 * @version 4.1.1
 */
var ApexRedirect = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ApexRedirect</code>.
   * @alias module:model/ApexRedirect
   * @implements module:model/ServiceIdAndVersion
   * @implements module:model/Timestamps
   * @implements module:model/ApexRedirectAllOf
   */
  function ApexRedirect() {
    _classCallCheck(this, ApexRedirect);
    _ServiceIdAndVersion["default"].initialize(this);
    _Timestamps["default"].initialize(this);
    _ApexRedirectAllOf["default"].initialize(this);
    ApexRedirect.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(ApexRedirect, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ApexRedirect</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ApexRedirect} obj Optional instance to populate.
     * @return {module:model/ApexRedirect} The populated <code>ApexRedirect</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ApexRedirect();
        _ServiceIdAndVersion["default"].constructFromObject(data, obj);
        _Timestamps["default"].constructFromObject(data, obj);
        _ApexRedirectAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('service_id')) {
          obj['service_id'] = _ApiClient["default"].convertToType(data['service_id'], 'String');
        }
        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('deleted_at')) {
          obj['deleted_at'] = _ApiClient["default"].convertToType(data['deleted_at'], 'Date');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
        if (data.hasOwnProperty('status_code')) {
          obj['status_code'] = _ApiClient["default"].convertToType(data['status_code'], 'Number');
        }
        if (data.hasOwnProperty('domains')) {
          obj['domains'] = _ApiClient["default"].convertToType(data['domains'], ['String']);
        }
        if (data.hasOwnProperty('feature_revision')) {
          obj['feature_revision'] = _ApiClient["default"].convertToType(data['feature_revision'], 'Number');
        }
      }
      return obj;
    }
  }]);
  return ApexRedirect;
}();
/**
 * @member {String} service_id
 */
ApexRedirect.prototype['service_id'] = undefined;

/**
 * @member {Number} version
 */
ApexRedirect.prototype['version'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
ApexRedirect.prototype['created_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */
ApexRedirect.prototype['deleted_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
ApexRedirect.prototype['updated_at'] = undefined;

/**
 * HTTP status code used to redirect the client.
 * @member {module:model/ApexRedirect.StatusCodeEnum} status_code
 */
ApexRedirect.prototype['status_code'] = undefined;

/**
 * Array of apex domains that should redirect to their WWW subdomain.
 * @member {Array.<String>} domains
 */
ApexRedirect.prototype['domains'] = undefined;

/**
 * Revision number of the apex redirect feature implementation. Defaults to the most recent revision.
 * @member {Number} feature_revision
 */
ApexRedirect.prototype['feature_revision'] = undefined;

// Implement ServiceIdAndVersion interface:
/**
 * @member {String} service_id
 */
_ServiceIdAndVersion["default"].prototype['service_id'] = undefined;
/**
 * @member {Number} version
 */
_ServiceIdAndVersion["default"].prototype['version'] = undefined;
// Implement Timestamps interface:
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
_Timestamps["default"].prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */
_Timestamps["default"].prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
_Timestamps["default"].prototype['updated_at'] = undefined;
// Implement ApexRedirectAllOf interface:
/**
 * HTTP status code used to redirect the client.
 * @member {module:model/ApexRedirectAllOf.StatusCodeEnum} status_code
 */
_ApexRedirectAllOf["default"].prototype['status_code'] = undefined;
/**
 * Array of apex domains that should redirect to their WWW subdomain.
 * @member {Array.<String>} domains
 */
_ApexRedirectAllOf["default"].prototype['domains'] = undefined;
/**
 * Revision number of the apex redirect feature implementation. Defaults to the most recent revision.
 * @member {Number} feature_revision
 */
_ApexRedirectAllOf["default"].prototype['feature_revision'] = undefined;

/**
 * Allowed values for the <code>status_code</code> property.
 * @enum {Number}
 * @readonly
 */
ApexRedirect['StatusCodeEnum'] = {
  /**
   * value: 301
   * @const
   */
  "301": 301,
  /**
   * value: 302
   * @const
   */
  "302": 302,
  /**
   * value: 307
   * @const
   */
  "307": 307,
  /**
   * value: 308
   * @const
   */
  "308": 308
};
var _default = ApexRedirect;
exports["default"] = _default;
