// fastly/dist/model/BillingEstimateResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Billing = _interopRequireDefault(require("./Billing"));
var _BillingEstimateResponseAllOf = _interopRequireDefault(require("./BillingEstimateResponseAllOf"));
var _BillingEstimateResponseAllOfLines = _interopRequireDefault(require("./BillingEstimateResponseAllOfLines"));
var _BillingStatus = _interopRequireDefault(require("./BillingStatus"));
var _BillingTotal = _interopRequireDefault(require("./BillingTotal"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The BillingEstimateResponse model module.
 * @module model/BillingEstimateResponse
 * @version 4.1.1
 */
var BillingEstimateResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>BillingEstimateResponse</code>.
   * @alias module:model/BillingEstimateResponse
   * @implements module:model/Billing
   * @implements module:model/BillingEstimateResponseAllOf
   */
  function BillingEstimateResponse() {
    _classCallCheck(this, BillingEstimateResponse);
    _Billing["default"].initialize(this);
    _BillingEstimateResponseAllOf["default"].initialize(this);
    BillingEstimateResponse.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(BillingEstimateResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>BillingEstimateResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BillingEstimateResponse} obj Optional instance to populate.
     * @return {module:model/BillingEstimateResponse} The populated <code>BillingEstimateResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BillingEstimateResponse();
        _Billing["default"].constructFromObject(data, obj);
        _BillingEstimateResponseAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('end_time')) {
          obj['end_time'] = _ApiClient["default"].convertToType(data['end_time'], 'Date');
        }
        if (data.hasOwnProperty('start_time')) {
          obj['start_time'] = _ApiClient["default"].convertToType(data['start_time'], 'Date');
        }
        if (data.hasOwnProperty('invoice_id')) {
          obj['invoice_id'] = _ApiClient["default"].convertToType(data['invoice_id'], 'String');
        }
        if (data.hasOwnProperty('customer_id')) {
          obj['customer_id'] = _ApiClient["default"].convertToType(data['customer_id'], 'String');
        }
        if (data.hasOwnProperty('vendor_state')) {
          obj['vendor_state'] = _ApiClient["default"].convertToType(data['vendor_state'], 'String');
        }
        if (data.hasOwnProperty('status')) {
          obj['status'] = _BillingStatus["default"].constructFromObject(data['status']);
        }
        if (data.hasOwnProperty('total')) {
          obj['total'] = _BillingTotal["default"].constructFromObject(data['total']);
        }
        if (data.hasOwnProperty('regions')) {
          obj['regions'] = _ApiClient["default"].convertToType(data['regions'], {
            'String': {
              'String': Object
            }
          });
        }
        if (data.hasOwnProperty('lines')) {
          obj['lines'] = _ApiClient["default"].convertToType(data['lines'], [_BillingEstimateResponseAllOfLines["default"]]);
        }
      }
      return obj;
    }
  }]);
  return BillingEstimateResponse;
}();
/**
 * Date and time in ISO 8601 format.
 * @member {Date} end_time
 */
BillingEstimateResponse.prototype['end_time'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} start_time
 */
BillingEstimateResponse.prototype['start_time'] = undefined;

/**
 * @member {String} invoice_id
 */
BillingEstimateResponse.prototype['invoice_id'] = undefined;

/**
 * @member {String} customer_id
 */
BillingEstimateResponse.prototype['customer_id'] = undefined;

/**
 * The current state of our third-party billing vendor. One of `up` or `down`.
 * @member {String} vendor_state
 */
BillingEstimateResponse.prototype['vendor_state'] = undefined;

/**
 * @member {module:model/BillingStatus} status
 */
BillingEstimateResponse.prototype['status'] = undefined;

/**
 * @member {module:model/BillingTotal} total
 */
BillingEstimateResponse.prototype['total'] = undefined;

/**
 * Breakdown of regional data for products that are region based.
 * @member {Object.<String, Object.<String, Object>>} regions
 */
BillingEstimateResponse.prototype['regions'] = undefined;

/**
 * @member {Array.<module:model/BillingEstimateResponseAllOfLines>} lines
 */
BillingEstimateResponse.prototype['lines'] = undefined;

// Implement Billing interface:
/**
 * Date and time in ISO 8601 format.
 * @member {Date} end_time
 */
_Billing["default"].prototype['end_time'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} start_time
 */
_Billing["default"].prototype['start_time'] = undefined;
/**
 * @member {String} invoice_id
 */
_Billing["default"].prototype['invoice_id'] = undefined;
/**
 * @member {String} customer_id
 */
_Billing["default"].prototype['customer_id'] = undefined;
/**
 * The current state of our third-party billing vendor. One of `up` or `down`.
 * @member {String} vendor_state
 */
_Billing["default"].prototype['vendor_state'] = undefined;
/**
 * @member {module:model/BillingStatus} status
 */
_Billing["default"].prototype['status'] = undefined;
/**
 * @member {module:model/BillingTotal} total
 */
_Billing["default"].prototype['total'] = undefined;
/**
 * Breakdown of regional data for products that are region based.
 * @member {Object.<String, Object.<String, Object>>} regions
 */
_Billing["default"].prototype['regions'] = undefined;
// Implement BillingEstimateResponseAllOf interface:
/**
 * @member {Array.<module:model/BillingEstimateResponseAllOfLines>} lines
 */
_BillingEstimateResponseAllOf["default"].prototype['lines'] = undefined;
var _default = BillingEstimateResponse;
exports["default"] = _default;
