// fastly/dist/model/ConditionResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Condition = _interopRequireDefault(require("./Condition"));
var _Timestamps = _interopRequireDefault(require("./Timestamps"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The ConditionResponse model module.
 * @module model/ConditionResponse
 * @version 4.1.1
 */
var ConditionResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ConditionResponse</code>.
   * @alias module:model/ConditionResponse
   * @implements module:model/Condition
   * @implements module:model/Timestamps
   */
  function ConditionResponse() {
    _classCallCheck(this, ConditionResponse);
    _Condition["default"].initialize(this);
    _Timestamps["default"].initialize(this);
    ConditionResponse.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(ConditionResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ConditionResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ConditionResponse} obj Optional instance to populate.
     * @return {module:model/ConditionResponse} The populated <code>ConditionResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ConditionResponse();
        _Condition["default"].constructFromObject(data, obj);
        _Timestamps["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('priority')) {
          obj['priority'] = _ApiClient["default"].convertToType(data['priority'], 'String');
        }
        if (data.hasOwnProperty('statement')) {
          obj['statement'] = _ApiClient["default"].convertToType(data['statement'], 'String');
        }
        if (data.hasOwnProperty('service_id')) {
          obj['service_id'] = _ApiClient["default"].convertToType(data['service_id'], 'String');
        }
        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'String');
        }
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('deleted_at')) {
          obj['deleted_at'] = _ApiClient["default"].convertToType(data['deleted_at'], 'Date');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
      }
      return obj;
    }
  }]);
  return ConditionResponse;
}();
/**
 * A freeform descriptive note.
 * @member {String} comment
 */
ConditionResponse.prototype['comment'] = undefined;

/**
 * Name of the condition. Required.
 * @member {String} name
 */
ConditionResponse.prototype['name'] = undefined;

/**
 * A numeric string. Priority determines execution order. Lower numbers execute first.
 * @member {String} priority
 * @default '100'
 */
ConditionResponse.prototype['priority'] = '100';

/**
 * A conditional expression in VCL used to determine if the condition is met.
 * @member {String} statement
 */
ConditionResponse.prototype['statement'] = undefined;

/**
 * @member {String} service_id
 */
ConditionResponse.prototype['service_id'] = undefined;

/**
 * A numeric string that represents the service version.
 * @member {String} version
 */
ConditionResponse.prototype['version'] = undefined;

/**
 * Type of the condition. Required.
 * @member {module:model/ConditionResponse.TypeEnum} type
 */
ConditionResponse.prototype['type'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
ConditionResponse.prototype['created_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */
ConditionResponse.prototype['deleted_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
ConditionResponse.prototype['updated_at'] = undefined;

// Implement Condition interface:
/**
 * A freeform descriptive note.
 * @member {String} comment
 */
_Condition["default"].prototype['comment'] = undefined;
/**
 * Name of the condition. Required.
 * @member {String} name
 */
_Condition["default"].prototype['name'] = undefined;
/**
 * A numeric string. Priority determines execution order. Lower numbers execute first.
 * @member {String} priority
 * @default '100'
 */
_Condition["default"].prototype['priority'] = '100';
/**
 * A conditional expression in VCL used to determine if the condition is met.
 * @member {String} statement
 */
_Condition["default"].prototype['statement'] = undefined;
/**
 * @member {String} service_id
 */
_Condition["default"].prototype['service_id'] = undefined;
/**
 * A numeric string that represents the service version.
 * @member {String} version
 */
_Condition["default"].prototype['version'] = undefined;
/**
 * Type of the condition. Required.
 * @member {module:model/Condition.TypeEnum} type
 */
_Condition["default"].prototype['type'] = undefined;
// Implement Timestamps interface:
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
_Timestamps["default"].prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */
_Timestamps["default"].prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
_Timestamps["default"].prototype['updated_at'] = undefined;

/**
 * Allowed values for the <code>type</code> property.
 * @enum {String}
 * @readonly
 */
ConditionResponse['TypeEnum'] = {
  /**
   * value: "REQUEST"
   * @const
   */
  "REQUEST": "REQUEST",
  /**
   * value: "CACHE"
   * @const
   */
  "CACHE": "CACHE",
  /**
   * value: "RESPONSE"
   * @const
   */
  "RESPONSE": "RESPONSE",
  /**
   * value: "PREFETCH"
   * @const
   */
  "PREFETCH": "PREFETCH"
};
var _default = ConditionResponse;
exports["default"] = _default;
