// fastly/dist/model/DictionaryResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Dictionary = _interopRequireDefault(require("./Dictionary"));
var _DictionaryResponseAllOf = _interopRequireDefault(require("./DictionaryResponseAllOf"));
var _ServiceIdAndVersion = _interopRequireDefault(require("./ServiceIdAndVersion"));
var _Timestamps = _interopRequireDefault(require("./Timestamps"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The DictionaryResponse model module.
 * @module model/DictionaryResponse
 * @version 4.1.1
 */
var DictionaryResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DictionaryResponse</code>.
   * @alias module:model/DictionaryResponse
   * @implements module:model/Dictionary
   * @implements module:model/Timestamps
   * @implements module:model/ServiceIdAndVersion
   * @implements module:model/DictionaryResponseAllOf
   */
  function DictionaryResponse() {
    _classCallCheck(this, DictionaryResponse);
    _Dictionary["default"].initialize(this);
    _Timestamps["default"].initialize(this);
    _ServiceIdAndVersion["default"].initialize(this);
    _DictionaryResponseAllOf["default"].initialize(this);
    DictionaryResponse.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(DictionaryResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>DictionaryResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DictionaryResponse} obj Optional instance to populate.
     * @return {module:model/DictionaryResponse} The populated <code>DictionaryResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DictionaryResponse();
        _Dictionary["default"].constructFromObject(data, obj);
        _Timestamps["default"].constructFromObject(data, obj);
        _ServiceIdAndVersion["default"].constructFromObject(data, obj);
        _DictionaryResponseAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('write_only')) {
          obj['write_only'] = _ApiClient["default"].convertToType(data['write_only'], 'Boolean');
        }
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('deleted_at')) {
          obj['deleted_at'] = _ApiClient["default"].convertToType(data['deleted_at'], 'Date');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
        if (data.hasOwnProperty('service_id')) {
          obj['service_id'] = _ApiClient["default"].convertToType(data['service_id'], 'String');
        }
        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
      }
      return obj;
    }
  }]);
  return DictionaryResponse;
}();
/**
 * Name for the Dictionary (must start with an alphabetic character and can contain only alphanumeric characters, underscores, and whitespace).
 * @member {String} name
 */
DictionaryResponse.prototype['name'] = undefined;

/**
 * Determines if items in the dictionary are readable or not.
 * @member {Boolean} write_only
 * @default false
 */
DictionaryResponse.prototype['write_only'] = false;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
DictionaryResponse.prototype['created_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */
DictionaryResponse.prototype['deleted_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
DictionaryResponse.prototype['updated_at'] = undefined;

/**
 * @member {String} service_id
 */
DictionaryResponse.prototype['service_id'] = undefined;

/**
 * @member {Number} version
 */
DictionaryResponse.prototype['version'] = undefined;

/**
 * @member {String} id
 */
DictionaryResponse.prototype['id'] = undefined;

// Implement Dictionary interface:
/**
 * Name for the Dictionary (must start with an alphabetic character and can contain only alphanumeric characters, underscores, and whitespace).
 * @member {String} name
 */
_Dictionary["default"].prototype['name'] = undefined;
/**
 * Determines if items in the dictionary are readable or not.
 * @member {Boolean} write_only
 * @default false
 */
_Dictionary["default"].prototype['write_only'] = false;
// Implement Timestamps interface:
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
_Timestamps["default"].prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */
_Timestamps["default"].prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
_Timestamps["default"].prototype['updated_at'] = undefined;
// Implement ServiceIdAndVersion interface:
/**
 * @member {String} service_id
 */
_ServiceIdAndVersion["default"].prototype['service_id'] = undefined;
/**
 * @member {Number} version
 */
_ServiceIdAndVersion["default"].prototype['version'] = undefined;
// Implement DictionaryResponseAllOf interface:
/**
 * @member {String} id
 */
_DictionaryResponseAllOf["default"].prototype['id'] = undefined;
var _default = DictionaryResponse;
exports["default"] = _default;
