// fastly/dist/model/DictionaryItemResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _DictionaryItem = _interopRequireDefault(require("./DictionaryItem"));
var _DictionaryItemResponseAllOf = _interopRequireDefault(require("./DictionaryItemResponseAllOf"));
var _Timestamps = _interopRequireDefault(require("./Timestamps"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The DictionaryItemResponse model module.
 * @module model/DictionaryItemResponse
 * @version 4.1.1
 */
var DictionaryItemResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DictionaryItemResponse</code>.
   * @alias module:model/DictionaryItemResponse
   * @implements module:model/DictionaryItem
   * @implements module:model/Timestamps
   * @implements module:model/DictionaryItemResponseAllOf
   */
  function DictionaryItemResponse() {
    _classCallCheck(this, DictionaryItemResponse);
    _DictionaryItem["default"].initialize(this);
    _Timestamps["default"].initialize(this);
    _DictionaryItemResponseAllOf["default"].initialize(this);
    DictionaryItemResponse.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(DictionaryItemResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>DictionaryItemResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DictionaryItemResponse} obj Optional instance to populate.
     * @return {module:model/DictionaryItemResponse} The populated <code>DictionaryItemResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DictionaryItemResponse();
        _DictionaryItem["default"].constructFromObject(data, obj);
        _Timestamps["default"].constructFromObject(data, obj);
        _DictionaryItemResponseAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('item_key')) {
          obj['item_key'] = _ApiClient["default"].convertToType(data['item_key'], 'String');
        }
        if (data.hasOwnProperty('item_value')) {
          obj['item_value'] = _ApiClient["default"].convertToType(data['item_value'], 'String');
        }
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('deleted_at')) {
          obj['deleted_at'] = _ApiClient["default"].convertToType(data['deleted_at'], 'Date');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
        if (data.hasOwnProperty('dictionary_id')) {
          obj['dictionary_id'] = _ApiClient["default"].convertToType(data['dictionary_id'], 'String');
        }
        if (data.hasOwnProperty('service_id')) {
          obj['service_id'] = _ApiClient["default"].convertToType(data['service_id'], 'String');
        }
      }
      return obj;
    }
  }]);
  return DictionaryItemResponse;
}();
/**
 * Item key, maximum 256 characters.
 * @member {String} item_key
 */
DictionaryItemResponse.prototype['item_key'] = undefined;

/**
 * Item value, maximum 8000 characters.
 * @member {String} item_value
 */
DictionaryItemResponse.prototype['item_value'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
DictionaryItemResponse.prototype['created_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */
DictionaryItemResponse.prototype['deleted_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
DictionaryItemResponse.prototype['updated_at'] = undefined;

/**
 * @member {String} dictionary_id
 */
DictionaryItemResponse.prototype['dictionary_id'] = undefined;

/**
 * @member {String} service_id
 */
DictionaryItemResponse.prototype['service_id'] = undefined;

// Implement DictionaryItem interface:
/**
 * Item key, maximum 256 characters.
 * @member {String} item_key
 */
_DictionaryItem["default"].prototype['item_key'] = undefined;
/**
 * Item value, maximum 8000 characters.
 * @member {String} item_value
 */
_DictionaryItem["default"].prototype['item_value'] = undefined;
// Implement Timestamps interface:
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
_Timestamps["default"].prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */
_Timestamps["default"].prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
_Timestamps["default"].prototype['updated_at'] = undefined;
// Implement DictionaryItemResponseAllOf interface:
/**
 * @member {String} dictionary_id
 */
_DictionaryItemResponseAllOf["default"].prototype['dictionary_id'] = undefined;
/**
 * @member {String} service_id
 */
_DictionaryItemResponseAllOf["default"].prototype['service_id'] = undefined;
var _default = DictionaryItemResponse;
exports["default"] = _default;
