// fastly/dist/model/HttpResponseFormat.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The HttpResponseFormat model module.
 * @module model/HttpResponseFormat
 * @version 4.1.1
 */
var HttpResponseFormat = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>HttpResponseFormat</code>.
   * Payload format for delivering to subscribers of whole HTTP responses (&#x60;response&#x60; hold mode). One of &#x60;body&#x60; or &#x60;body-bin&#x60; must be specified.
   * @alias module:model/HttpResponseFormat
   */
  function HttpResponseFormat() {
    _classCallCheck(this, HttpResponseFormat);
    HttpResponseFormat.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(HttpResponseFormat, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>HttpResponseFormat</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/HttpResponseFormat} obj Optional instance to populate.
     * @return {module:model/HttpResponseFormat} The populated <code>HttpResponseFormat</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new HttpResponseFormat();
        if (data.hasOwnProperty('code')) {
          obj['code'] = _ApiClient["default"].convertToType(data['code'], 'Number');
        }
        if (data.hasOwnProperty('reason')) {
          obj['reason'] = _ApiClient["default"].convertToType(data['reason'], 'String');
        }
        if (data.hasOwnProperty('headers')) {
          obj['headers'] = _ApiClient["default"].convertToType(data['headers'], {
            'String': 'String'
          });
        }
        if (data.hasOwnProperty('body')) {
          obj['body'] = _ApiClient["default"].convertToType(data['body'], 'String');
        }
        if (data.hasOwnProperty('body-bin')) {
          obj['body-bin'] = _ApiClient["default"].convertToType(data['body-bin'], 'String');
        }
      }
      return obj;
    }
  }]);
  return HttpResponseFormat;
}();
/**
 * The HTTP status code.
 * @member {Number} code
 * @default 200
 */
HttpResponseFormat.prototype['code'] = 200;

/**
 * The HTTP status string. Defaults to a string appropriate for `code`.
 * @member {String} reason
 */
HttpResponseFormat.prototype['reason'] = undefined;

/**
 * A map of arbitrary HTTP response headers to include in the response.
 * @member {Object.<String, String>} headers
 */
HttpResponseFormat.prototype['headers'] = undefined;

/**
 * The response body as a string.
 * @member {String} body
 */
HttpResponseFormat.prototype['body'] = undefined;

/**
 * The response body as a base64-encoded binary blob.
 * @member {String} body-bin
 */
HttpResponseFormat.prototype['body-bin'] = undefined;
var _default = HttpResponseFormat;
exports["default"] = _default;
