// fastly/dist/model/IamRole.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _IamRoleAllOf = _interopRequireDefault(require("./IamRoleAllOf"));
var _TimestampsNoDelete = _interopRequireDefault(require("./TimestampsNoDelete"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The IamRole model module.
 * @module model/IamRole
 * @version 4.1.1
 */
var IamRole = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>IamRole</code>.
   * @alias module:model/IamRole
   * @implements module:model/TimestampsNoDelete
   * @implements module:model/IamRoleAllOf
   */
  function IamRole() {
    _classCallCheck(this, IamRole);
    _TimestampsNoDelete["default"].initialize(this);
    _IamRoleAllOf["default"].initialize(this);
    IamRole.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(IamRole, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>IamRole</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/IamRole} obj Optional instance to populate.
     * @return {module:model/IamRole} The populated <code>IamRole</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new IamRole();
        _TimestampsNoDelete["default"].constructFromObject(data, obj);
        _IamRoleAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('object')) {
          obj['object'] = _ApiClient["default"].convertToType(data['object'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }
        if (data.hasOwnProperty('custom')) {
          obj['custom'] = _ApiClient["default"].convertToType(data['custom'], 'Boolean');
        }
        if (data.hasOwnProperty('permissions_count')) {
          obj['permissions_count'] = _ApiClient["default"].convertToType(data['permissions_count'], 'Number');
        }
      }
      return obj;
    }
  }]);
  return IamRole;
}();
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
IamRole.prototype['created_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
IamRole.prototype['updated_at'] = undefined;

/**
 * Alphanumeric string identifying the role.
 * @member {String} id
 */
IamRole.prototype['id'] = undefined;

/**
 * The type of the object.
 * @member {String} object
 */
IamRole.prototype['object'] = undefined;

/**
 * Name of the role.
 * @member {String} name
 */
IamRole.prototype['name'] = undefined;

/**
 * Description of the role.
 * @member {String} description
 */
IamRole.prototype['description'] = undefined;

/**
 * This attribute is set to `true` if the role is managed by the customer. It is set to `false` if the role was created by Fastly.
 * @member {Boolean} custom
 */
IamRole.prototype['custom'] = undefined;

/**
 * Number of permissions assigned to the role.
 * @member {Number} permissions_count
 */
IamRole.prototype['permissions_count'] = undefined;

// Implement TimestampsNoDelete interface:
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
_TimestampsNoDelete["default"].prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
_TimestampsNoDelete["default"].prototype['updated_at'] = undefined;
// Implement IamRoleAllOf interface:
/**
 * Alphanumeric string identifying the role.
 * @member {String} id
 */
_IamRoleAllOf["default"].prototype['id'] = undefined;
/**
 * The type of the object.
 * @member {String} object
 */
_IamRoleAllOf["default"].prototype['object'] = undefined;
/**
 * Name of the role.
 * @member {String} name
 */
_IamRoleAllOf["default"].prototype['name'] = undefined;
/**
 * Description of the role.
 * @member {String} description
 */
_IamRoleAllOf["default"].prototype['description'] = undefined;
/**
 * This attribute is set to `true` if the role is managed by the customer. It is set to `false` if the role was created by Fastly.
 * @member {Boolean} custom
 */
_IamRoleAllOf["default"].prototype['custom'] = undefined;
/**
 * Number of permissions assigned to the role.
 * @member {Number} permissions_count
 */
_IamRoleAllOf["default"].prototype['permissions_count'] = undefined;
var _default = IamRole;
exports["default"] = _default;
