// fastly/dist/model/IamServiceGroup.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _IamServiceGroupAllOf = _interopRequireDefault(require("./IamServiceGroupAllOf"));
var _TimestampsNoDelete = _interopRequireDefault(require("./TimestampsNoDelete"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The IamServiceGroup model module.
 * @module model/IamServiceGroup
 * @version 4.1.1
 */
var IamServiceGroup = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>IamServiceGroup</code>.
   * @alias module:model/IamServiceGroup
   * @implements module:model/TimestampsNoDelete
   * @implements module:model/IamServiceGroupAllOf
   */
  function IamServiceGroup() {
    _classCallCheck(this, IamServiceGroup);
    _TimestampsNoDelete["default"].initialize(this);
    _IamServiceGroupAllOf["default"].initialize(this);
    IamServiceGroup.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(IamServiceGroup, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>IamServiceGroup</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/IamServiceGroup} obj Optional instance to populate.
     * @return {module:model/IamServiceGroup} The populated <code>IamServiceGroup</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new IamServiceGroup();
        _TimestampsNoDelete["default"].constructFromObject(data, obj);
        _IamServiceGroupAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('object')) {
          obj['object'] = _ApiClient["default"].convertToType(data['object'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }
        if (data.hasOwnProperty('services_count')) {
          obj['services_count'] = _ApiClient["default"].convertToType(data['services_count'], 'Number');
        }
      }
      return obj;
    }
  }]);
  return IamServiceGroup;
}();
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
IamServiceGroup.prototype['created_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
IamServiceGroup.prototype['updated_at'] = undefined;

/**
 * Alphanumeric string identifying the service group.
 * @member {String} id
 */
IamServiceGroup.prototype['id'] = undefined;

/**
 * The type of the object.
 * @member {String} object
 */
IamServiceGroup.prototype['object'] = undefined;

/**
 * Name of the service group.
 * @member {String} name
 */
IamServiceGroup.prototype['name'] = undefined;

/**
 * Description of the service group.
 * @member {String} description
 */
IamServiceGroup.prototype['description'] = undefined;

/**
 * Number of services in the service group.
 * @member {Number} services_count
 */
IamServiceGroup.prototype['services_count'] = undefined;

// Implement TimestampsNoDelete interface:
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
_TimestampsNoDelete["default"].prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
_TimestampsNoDelete["default"].prototype['updated_at'] = undefined;
// Implement IamServiceGroupAllOf interface:
/**
 * Alphanumeric string identifying the service group.
 * @member {String} id
 */
_IamServiceGroupAllOf["default"].prototype['id'] = undefined;
/**
 * The type of the object.
 * @member {String} object
 */
_IamServiceGroupAllOf["default"].prototype['object'] = undefined;
/**
 * Name of the service group.
 * @member {String} name
 */
_IamServiceGroupAllOf["default"].prototype['name'] = undefined;
/**
 * Description of the service group.
 * @member {String} description
 */
_IamServiceGroupAllOf["default"].prototype['description'] = undefined;
/**
 * Number of services in the service group.
 * @member {Number} services_count
 */
_IamServiceGroupAllOf["default"].prototype['services_count'] = undefined;
var _default = IamServiceGroup;
exports["default"] = _default;
