// fastly/dist/model/IamUserGroup.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _IamUserGroupAllOf = _interopRequireDefault(require("./IamUserGroupAllOf"));
var _TimestampsNoDelete = _interopRequireDefault(require("./TimestampsNoDelete"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The IamUserGroup model module.
 * @module model/IamUserGroup
 * @version 4.1.1
 */
var IamUserGroup = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>IamUserGroup</code>.
   * @alias module:model/IamUserGroup
   * @implements module:model/TimestampsNoDelete
   * @implements module:model/IamUserGroupAllOf
   */
  function IamUserGroup() {
    _classCallCheck(this, IamUserGroup);
    _TimestampsNoDelete["default"].initialize(this);
    _IamUserGroupAllOf["default"].initialize(this);
    IamUserGroup.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(IamUserGroup, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>IamUserGroup</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/IamUserGroup} obj Optional instance to populate.
     * @return {module:model/IamUserGroup} The populated <code>IamUserGroup</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new IamUserGroup();
        _TimestampsNoDelete["default"].constructFromObject(data, obj);
        _IamUserGroupAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }
        if (data.hasOwnProperty('invitations_count')) {
          obj['invitations_count'] = _ApiClient["default"].convertToType(data['invitations_count'], 'Number');
        }
        if (data.hasOwnProperty('users_count')) {
          obj['users_count'] = _ApiClient["default"].convertToType(data['users_count'], 'Number');
        }
        if (data.hasOwnProperty('roles_count')) {
          obj['roles_count'] = _ApiClient["default"].convertToType(data['roles_count'], 'Number');
        }
      }
      return obj;
    }
  }]);
  return IamUserGroup;
}();
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
IamUserGroup.prototype['created_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
IamUserGroup.prototype['updated_at'] = undefined;

/**
 * Alphanumeric string identifying the user group.
 * @member {String} id
 */
IamUserGroup.prototype['id'] = undefined;

/**
 * Name of the user group.
 * @member {String} name
 */
IamUserGroup.prototype['name'] = undefined;

/**
 * Description of the user group.
 * @member {String} description
 */
IamUserGroup.prototype['description'] = undefined;

/**
 * Number of invitations added to the user group.
 * @member {Number} invitations_count
 */
IamUserGroup.prototype['invitations_count'] = undefined;

/**
 * Number of users added to the user group.
 * @member {Number} users_count
 */
IamUserGroup.prototype['users_count'] = undefined;

/**
 * Number of roles added to the user group.
 * @member {Number} roles_count
 */
IamUserGroup.prototype['roles_count'] = undefined;

// Implement TimestampsNoDelete interface:
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
_TimestampsNoDelete["default"].prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
_TimestampsNoDelete["default"].prototype['updated_at'] = undefined;
// Implement IamUserGroupAllOf interface:
/**
 * Alphanumeric string identifying the user group.
 * @member {String} id
 */
_IamUserGroupAllOf["default"].prototype['id'] = undefined;
/**
 * Name of the user group.
 * @member {String} name
 */
_IamUserGroupAllOf["default"].prototype['name'] = undefined;
/**
 * Description of the user group.
 * @member {String} description
 */
_IamUserGroupAllOf["default"].prototype['description'] = undefined;
/**
 * Number of invitations added to the user group.
 * @member {Number} invitations_count
 */
_IamUserGroupAllOf["default"].prototype['invitations_count'] = undefined;
/**
 * Number of users added to the user group.
 * @member {Number} users_count
 */
_IamUserGroupAllOf["default"].prototype['users_count'] = undefined;
/**
 * Number of roles added to the user group.
 * @member {Number} roles_count
 */
_IamUserGroupAllOf["default"].prototype['roles_count'] = undefined;
var _default = IamUserGroup;
exports["default"] = _default;
