// fastly/dist/model/PackageMetadata.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The PackageMetadata model module.
 * @module model/PackageMetadata
 * @version 4.1.1
 */
var PackageMetadata = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PackageMetadata</code>.
   * [Package metadata](#metadata-model) that has been extracted from the uploaded package. 
   * @alias module:model/PackageMetadata
   */
  function PackageMetadata() {
    _classCallCheck(this, PackageMetadata);
    PackageMetadata.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(PackageMetadata, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>PackageMetadata</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PackageMetadata} obj Optional instance to populate.
     * @return {module:model/PackageMetadata} The populated <code>PackageMetadata</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PackageMetadata();
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('description')) {
          obj['description'] = _ApiClient["default"].convertToType(data['description'], 'String');
        }
        if (data.hasOwnProperty('authors')) {
          obj['authors'] = _ApiClient["default"].convertToType(data['authors'], ['String']);
        }
        if (data.hasOwnProperty('language')) {
          obj['language'] = _ApiClient["default"].convertToType(data['language'], 'String');
        }
        if (data.hasOwnProperty('size')) {
          obj['size'] = _ApiClient["default"].convertToType(data['size'], 'Number');
        }
        if (data.hasOwnProperty('hashsum')) {
          obj['hashsum'] = _ApiClient["default"].convertToType(data['hashsum'], 'String');
        }
        if (data.hasOwnProperty('files_hash')) {
          obj['files_hash'] = _ApiClient["default"].convertToType(data['files_hash'], 'String');
        }
      }
      return obj;
    }
  }]);
  return PackageMetadata;
}();
/**
 * Name of the Compute@Edge package.
 * @member {String} name
 */
PackageMetadata.prototype['name'] = undefined;

/**
 * Description of the Compute@Edge package.
 * @member {String} description
 */
PackageMetadata.prototype['description'] = undefined;

/**
 * A list of package authors' email addresses.
 * @member {Array.<String>} authors
 */
PackageMetadata.prototype['authors'] = undefined;

/**
 * The language of the Compute@Edge package.
 * @member {String} language
 */
PackageMetadata.prototype['language'] = undefined;

/**
 * Size of the Compute@Edge package in bytes.
 * @member {Number} size
 */
PackageMetadata.prototype['size'] = undefined;

/**
 * Hash of the Compute@Edge package.
 * @member {String} hashsum
 */
PackageMetadata.prototype['hashsum'] = undefined;

/**
 * Hash of the files within the Compute@Edge package.
 * @member {String} files_hash
 */
PackageMetadata.prototype['files_hash'] = undefined;
var _default = PackageMetadata;
exports["default"] = _default;
