// fastly/dist/model/Pop.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _PopCoordinates = _interopRequireDefault(require("./PopCoordinates"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The Pop model module.
 * @module model/Pop
 * @version 4.1.1
 */
var Pop = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Pop</code>.
   * @alias module:model/Pop
   * @param code {String} the three-letter code for the [POP](https://developer.fastly.com/learning/concepts/pop/)
   * @param name {String} the name of the POP
   * @param group {String} 
   * @param region {module:model/Pop.RegionEnum} 
   * @param stats_region {module:model/Pop.StatsRegionEnum} the region used for stats reporting
   * @param billing_region {module:model/Pop.BillingRegionEnum} the region used for billing
   */
  function Pop(code, name, group, region, stats_region, billing_region) {
    _classCallCheck(this, Pop);
    Pop.initialize(this, code, name, group, region, stats_region, billing_region);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Pop, null, [{
    key: "initialize",
    value: function initialize(obj, code, name, group, region, stats_region, billing_region) {
      obj['code'] = code;
      obj['name'] = name;
      obj['group'] = group;
      obj['region'] = region;
      obj['stats_region'] = stats_region;
      obj['billing_region'] = billing_region;
    }

    /**
     * Constructs a <code>Pop</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Pop} obj Optional instance to populate.
     * @return {module:model/Pop} The populated <code>Pop</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Pop();
        if (data.hasOwnProperty('code')) {
          obj['code'] = _ApiClient["default"].convertToType(data['code'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('group')) {
          obj['group'] = _ApiClient["default"].convertToType(data['group'], 'String');
        }
        if (data.hasOwnProperty('region')) {
          obj['region'] = _ApiClient["default"].convertToType(data['region'], 'String');
        }
        if (data.hasOwnProperty('stats_region')) {
          obj['stats_region'] = _ApiClient["default"].convertToType(data['stats_region'], 'String');
        }
        if (data.hasOwnProperty('billing_region')) {
          obj['billing_region'] = _ApiClient["default"].convertToType(data['billing_region'], 'String');
        }
        if (data.hasOwnProperty('coordinates')) {
          obj['coordinates'] = _PopCoordinates["default"].constructFromObject(data['coordinates']);
        }
        if (data.hasOwnProperty('shield')) {
          obj['shield'] = _ApiClient["default"].convertToType(data['shield'], 'String');
        }
      }
      return obj;
    }
  }]);
  return Pop;
}();
/**
 * the three-letter code for the [POP](https://developer.fastly.com/learning/concepts/pop/)
 * @member {String} code
 */
Pop.prototype['code'] = undefined;

/**
 * the name of the POP
 * @member {String} name
 */
Pop.prototype['name'] = undefined;

/**
 * @member {String} group
 */
Pop.prototype['group'] = undefined;

/**
 * @member {module:model/Pop.RegionEnum} region
 */
Pop.prototype['region'] = undefined;

/**
 * the region used for stats reporting
 * @member {module:model/Pop.StatsRegionEnum} stats_region
 */
Pop.prototype['stats_region'] = undefined;

/**
 * the region used for billing
 * @member {module:model/Pop.BillingRegionEnum} billing_region
 */
Pop.prototype['billing_region'] = undefined;

/**
 * @member {module:model/PopCoordinates} coordinates
 */
Pop.prototype['coordinates'] = undefined;

/**
 * the name of the [shield code](https://developer.fastly.com/learning/concepts/shielding/#choosing-a-shield-location) if this POP is suitable for shielding
 * @member {String} shield
 */
Pop.prototype['shield'] = undefined;

/**
 * Allowed values for the <code>region</code> property.
 * @enum {String}
 * @readonly
 */
Pop['RegionEnum'] = {
  /**
   * value: "APAC"
   * @const
   */
  "APAC": "APAC",
  /**
   * value: "Asia"
   * @const
   */
  "Asia": "Asia",
  /**
   * value: "AF-West"
   * @const
   */
  "AF-West": "AF-West",
  /**
   * value: "EU-Central"
   * @const
   */
  "EU-Central": "EU-Central",
  /**
   * value: "EU-East"
   * @const
   */
  "EU-East": "EU-East",
  /**
   * value: "EU-West"
   * @const
   */
  "EU-West": "EU-West",
  /**
   * value: "Middle-East"
   * @const
   */
  "Middle-East": "Middle-East",
  /**
   * value: "North-America"
   * @const
   */
  "North-America": "North-America",
  /**
   * value: "SA-South"
   * @const
   */
  "SA-South": "SA-South",
  /**
   * value: "SA-East"
   * @const
   */
  "SA-East": "SA-East",
  /**
   * value: "SA-West"
   * @const
   */
  "SA-West": "SA-West",
  /**
   * value: "SA-North"
   * @const
   */
  "SA-North": "SA-North",
  /**
   * value: "South-Africa"
   * @const
   */
  "South-Africa": "South-Africa",
  /**
   * value: "South-America"
   * @const
   */
  "South-America": "South-America",
  /**
   * value: "US-Central"
   * @const
   */
  "US-Central": "US-Central",
  /**
   * value: "US-East"
   * @const
   */
  "US-East": "US-East",
  /**
   * value: "US-West"
   * @const
   */
  "US-West": "US-West",
  /**
   * value: "Asia-South"
   * @const
   */
  "Asia-South": "Asia-South"
};

/**
 * Allowed values for the <code>stats_region</code> property.
 * @enum {String}
 * @readonly
 */
Pop['StatsRegionEnum'] = {
  /**
   * value: "southamerica_std"
   * @const
   */
  "southamerica_std": "southamerica_std",
  /**
   * value: "africa_std"
   * @const
   */
  "africa_std": "africa_std",
  /**
   * value: "anzac"
   * @const
   */
  "anzac": "anzac",
  /**
   * value: "asia"
   * @const
   */
  "asia": "asia",
  /**
   * value: "europe"
   * @const
   */
  "europe": "europe",
  /**
   * value: "usa"
   * @const
   */
  "usa": "usa",
  /**
   * value: "asia_india"
   * @const
   */
  "asia_india": "asia_india",
  /**
   * value: "asia_southkorea"
   * @const
   */
  "asia_southkorea": "asia_southkorea"
};

/**
 * Allowed values for the <code>billing_region</code> property.
 * @enum {String}
 * @readonly
 */
Pop['BillingRegionEnum'] = {
  /**
   * value: "Africa"
   * @const
   */
  "Africa": "Africa",
  /**
   * value: "Australia"
   * @const
   */
  "Australia": "Australia",
  /**
   * value: "Asia"
   * @const
   */
  "Asia": "Asia",
  /**
   * value: "Europe"
   * @const
   */
  "Europe": "Europe",
  /**
   * value: "India"
   * @const
   */
  "India": "India",
  /**
   * value: "North America"
   * @const
   */
  "North America": "North America",
  /**
   * value: "South Korea"
   * @const
   */
  "South Korea": "South Korea",
  /**
   * value: "South America"
   * @const
   */
  "South America": "South America"
};
var _default = Pop;
exports["default"] = _default;
