// fastly/dist/model/RelationshipsForWafRule.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _RelationshipWafRuleRevisionWafRuleRevisions = _interopRequireDefault(require("./RelationshipWafRuleRevisionWafRuleRevisions"));
var _RelationshipWafRuleRevisions = _interopRequireDefault(require("./RelationshipWafRuleRevisions"));
var _RelationshipWafTags = _interopRequireDefault(require("./RelationshipWafTags"));
var _RelationshipWafTagsWafTags = _interopRequireDefault(require("./RelationshipWafTagsWafTags"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The RelationshipsForWafRule model module.
 * @module model/RelationshipsForWafRule
 * @version 4.1.1
 */
var RelationshipsForWafRule = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>RelationshipsForWafRule</code>.
   * @alias module:model/RelationshipsForWafRule
   * @implements module:model/RelationshipWafTags
   * @implements module:model/RelationshipWafRuleRevisions
   */
  function RelationshipsForWafRule() {
    _classCallCheck(this, RelationshipsForWafRule);
    _RelationshipWafTags["default"].initialize(this);
    _RelationshipWafRuleRevisions["default"].initialize(this);
    RelationshipsForWafRule.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(RelationshipsForWafRule, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>RelationshipsForWafRule</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/RelationshipsForWafRule} obj Optional instance to populate.
     * @return {module:model/RelationshipsForWafRule} The populated <code>RelationshipsForWafRule</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new RelationshipsForWafRule();
        _RelationshipWafTags["default"].constructFromObject(data, obj);
        _RelationshipWafRuleRevisions["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('waf_tags')) {
          obj['waf_tags'] = _RelationshipWafTagsWafTags["default"].constructFromObject(data['waf_tags']);
        }
        if (data.hasOwnProperty('waf_rule_revisions')) {
          obj['waf_rule_revisions'] = _RelationshipWafRuleRevisionWafRuleRevisions["default"].constructFromObject(data['waf_rule_revisions']);
        }
      }
      return obj;
    }
  }]);
  return RelationshipsForWafRule;
}();
/**
 * @member {module:model/RelationshipWafTagsWafTags} waf_tags
 */
RelationshipsForWafRule.prototype['waf_tags'] = undefined;

/**
 * @member {module:model/RelationshipWafRuleRevisionWafRuleRevisions} waf_rule_revisions
 */
RelationshipsForWafRule.prototype['waf_rule_revisions'] = undefined;

// Implement RelationshipWafTags interface:
/**
 * @member {module:model/RelationshipWafTagsWafTags} waf_tags
 */
_RelationshipWafTags["default"].prototype['waf_tags'] = undefined;
// Implement RelationshipWafRuleRevisions interface:
/**
 * @member {module:model/RelationshipWafRuleRevisionWafRuleRevisions} waf_rule_revisions
 */
_RelationshipWafRuleRevisions["default"].prototype['waf_rule_revisions'] = undefined;
var _default = RelationshipsForWafRule;
exports["default"] = _default;
