// fastly/dist/model/SchemasSnippetResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Snippet = _interopRequireDefault(require("./Snippet"));
var _SnippetResponseAllOf = _interopRequireDefault(require("./SnippetResponseAllOf"));
var _Timestamps = _interopRequireDefault(require("./Timestamps"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The SchemasSnippetResponse model module.
 * @module model/SchemasSnippetResponse
 * @version 4.1.1
 */
var SchemasSnippetResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SchemasSnippetResponse</code>.
   * @alias module:model/SchemasSnippetResponse
   * @implements module:model/Snippet
   * @implements module:model/Timestamps
   * @implements module:model/SnippetResponseAllOf
   */
  function SchemasSnippetResponse() {
    _classCallCheck(this, SchemasSnippetResponse);
    _Snippet["default"].initialize(this);
    _Timestamps["default"].initialize(this);
    _SnippetResponseAllOf["default"].initialize(this);
    SchemasSnippetResponse.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(SchemasSnippetResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>SchemasSnippetResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SchemasSnippetResponse} obj Optional instance to populate.
     * @return {module:model/SchemasSnippetResponse} The populated <code>SchemasSnippetResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SchemasSnippetResponse();
        _Snippet["default"].constructFromObject(data, obj);
        _Timestamps["default"].constructFromObject(data, obj);
        _SnippetResponseAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('dynamic')) {
          obj['dynamic'] = _ApiClient["default"].convertToType(data['dynamic'], 'Number');
        }
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('content')) {
          obj['content'] = _ApiClient["default"].convertToType(data['content'], 'String');
        }
        if (data.hasOwnProperty('priority')) {
          obj['priority'] = _ApiClient["default"].convertToType(data['priority'], 'String');
        }
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('deleted_at')) {
          obj['deleted_at'] = _ApiClient["default"].convertToType(data['deleted_at'], 'Date');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
        if (data.hasOwnProperty('service_id')) {
          obj['service_id'] = _ApiClient["default"].convertToType(data['service_id'], 'String');
        }
        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'String');
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
      }
      return obj;
    }
  }]);
  return SchemasSnippetResponse;
}();
/**
 * The name for the snippet.
 * @member {String} name
 */
SchemasSnippetResponse.prototype['name'] = undefined;

/**
 * Sets the snippet version.
 * @member {module:model/SchemasSnippetResponse.DynamicEnum} dynamic
 */
SchemasSnippetResponse.prototype['dynamic'] = undefined;

/**
 * The location in generated VCL where the snippet should be placed.
 * @member {module:model/SchemasSnippetResponse.TypeEnum} type
 */
SchemasSnippetResponse.prototype['type'] = undefined;

/**
 * The VCL code that specifies exactly what the snippet does.
 * @member {String} content
 */
SchemasSnippetResponse.prototype['content'] = undefined;

/**
 * Priority determines execution order. Lower numbers execute first.
 * @member {String} priority
 * @default '100'
 */
SchemasSnippetResponse.prototype['priority'] = '100';

/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
SchemasSnippetResponse.prototype['created_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */
SchemasSnippetResponse.prototype['deleted_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
SchemasSnippetResponse.prototype['updated_at'] = undefined;

/**
 * @member {String} service_id
 */
SchemasSnippetResponse.prototype['service_id'] = undefined;

/**
 * String representing the number identifying a version of the service.
 * @member {String} version
 */
SchemasSnippetResponse.prototype['version'] = undefined;

/**
 * @member {String} id
 */
SchemasSnippetResponse.prototype['id'] = undefined;

// Implement Snippet interface:
/**
 * The name for the snippet.
 * @member {String} name
 */
_Snippet["default"].prototype['name'] = undefined;
/**
 * Sets the snippet version.
 * @member {module:model/Snippet.DynamicEnum} dynamic
 */
_Snippet["default"].prototype['dynamic'] = undefined;
/**
 * The location in generated VCL where the snippet should be placed.
 * @member {module:model/Snippet.TypeEnum} type
 */
_Snippet["default"].prototype['type'] = undefined;
/**
 * The VCL code that specifies exactly what the snippet does.
 * @member {String} content
 */
_Snippet["default"].prototype['content'] = undefined;
/**
 * Priority determines execution order. Lower numbers execute first.
 * @member {String} priority
 * @default '100'
 */
_Snippet["default"].prototype['priority'] = '100';
// Implement Timestamps interface:
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
_Timestamps["default"].prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */
_Timestamps["default"].prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
_Timestamps["default"].prototype['updated_at'] = undefined;
// Implement SnippetResponseAllOf interface:
/**
 * @member {String} service_id
 */
_SnippetResponseAllOf["default"].prototype['service_id'] = undefined;
/**
 * String representing the number identifying a version of the service.
 * @member {String} version
 */
_SnippetResponseAllOf["default"].prototype['version'] = undefined;
/**
 * @member {String} id
 */
_SnippetResponseAllOf["default"].prototype['id'] = undefined;

/**
 * Allowed values for the <code>dynamic</code> property.
 * @enum {Number}
 * @readonly
 */
SchemasSnippetResponse['DynamicEnum'] = {
  /**
   * value: 0
   * @const
   */
  "0": 0,
  /**
   * value: 1
   * @const
   */
  "1": 1
};

/**
 * Allowed values for the <code>type</code> property.
 * @enum {String}
 * @readonly
 */
SchemasSnippetResponse['TypeEnum'] = {
  /**
   * value: "init"
   * @const
   */
  "init": "init",
  /**
   * value: "recv"
   * @const
   */
  "recv": "recv",
  /**
   * value: "hash"
   * @const
   */
  "hash": "hash",
  /**
   * value: "hit"
   * @const
   */
  "hit": "hit",
  /**
   * value: "miss"
   * @const
   */
  "miss": "miss",
  /**
   * value: "pass"
   * @const
   */
  "pass": "pass",
  /**
   * value: "fetch"
   * @const
   */
  "fetch": "fetch",
  /**
   * value: "error"
   * @const
   */
  "error": "error",
  /**
   * value: "deliver"
   * @const
   */
  "deliver": "deliver",
  /**
   * value: "log"
   * @const
   */
  "log": "log",
  /**
   * value: "none"
   * @const
   */
  "none": "none"
};
var _default = SchemasSnippetResponse;
exports["default"] = _default;
