// fastly/dist/model/ServiceDetail.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _SchemasVersionResponse = _interopRequireDefault(require("./SchemasVersionResponse"));
var _ServiceDetailAllOf = _interopRequireDefault(require("./ServiceDetailAllOf"));
var _ServiceResponse = _interopRequireDefault(require("./ServiceResponse"));
var _ServiceVersionDetail = _interopRequireDefault(require("./ServiceVersionDetail"));
var _ServiceVersionDetailOrNull = _interopRequireDefault(require("./ServiceVersionDetailOrNull"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The ServiceDetail model module.
 * @module model/ServiceDetail
 * @version 4.1.1
 */
var ServiceDetail = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ServiceDetail</code>.
   * @alias module:model/ServiceDetail
   * @implements module:model/ServiceResponse
   * @implements module:model/ServiceDetailAllOf
   */
  function ServiceDetail() {
    _classCallCheck(this, ServiceDetail);
    _ServiceResponse["default"].initialize(this);
    _ServiceDetailAllOf["default"].initialize(this);
    ServiceDetail.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(ServiceDetail, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ServiceDetail</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ServiceDetail} obj Optional instance to populate.
     * @return {module:model/ServiceDetail} The populated <code>ServiceDetail</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ServiceDetail();
        _ServiceResponse["default"].constructFromObject(data, obj);
        _ServiceDetailAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('deleted_at')) {
          obj['deleted_at'] = _ApiClient["default"].convertToType(data['deleted_at'], 'Date');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
        if (data.hasOwnProperty('comment')) {
          obj['comment'] = _ApiClient["default"].convertToType(data['comment'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('customer_id')) {
          obj['customer_id'] = _ApiClient["default"].convertToType(data['customer_id'], 'String');
        }
        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }
        if (data.hasOwnProperty('publish_key')) {
          obj['publish_key'] = _ApiClient["default"].convertToType(data['publish_key'], 'String');
        }
        if (data.hasOwnProperty('paused')) {
          obj['paused'] = _ApiClient["default"].convertToType(data['paused'], 'Boolean');
        }
        if (data.hasOwnProperty('versions')) {
          obj['versions'] = _ApiClient["default"].convertToType(data['versions'], [_SchemasVersionResponse["default"]]);
        }
        if (data.hasOwnProperty('active_version')) {
          obj['active_version'] = _ServiceVersionDetailOrNull["default"].constructFromObject(data['active_version']);
        }
        if (data.hasOwnProperty('version')) {
          obj['version'] = _ServiceVersionDetail["default"].constructFromObject(data['version']);
        }
      }
      return obj;
    }
  }]);
  return ServiceDetail;
}();
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
ServiceDetail.prototype['created_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */
ServiceDetail.prototype['deleted_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
ServiceDetail.prototype['updated_at'] = undefined;

/**
 * A freeform descriptive note.
 * @member {String} comment
 */
ServiceDetail.prototype['comment'] = undefined;

/**
 * The name of the service.
 * @member {String} name
 */
ServiceDetail.prototype['name'] = undefined;

/**
 * Alphanumeric string identifying the customer.
 * @member {String} customer_id
 */
ServiceDetail.prototype['customer_id'] = undefined;

/**
 * The type of this service.
 * @member {module:model/ServiceDetail.TypeEnum} type
 */
ServiceDetail.prototype['type'] = undefined;

/**
 * @member {String} id
 */
ServiceDetail.prototype['id'] = undefined;

/**
 * Unused at this time.
 * @member {String} publish_key
 */
ServiceDetail.prototype['publish_key'] = undefined;

/**
 * Whether the service is paused. Services are paused due to a lack of traffic for an extended period of time. Services are resumed either when a draft version is activated or a locked version is cloned and reactivated.
 * @member {Boolean} paused
 */
ServiceDetail.prototype['paused'] = undefined;

/**
 * A list of [versions](/reference/api/services/version/) associated with the service.
 * @member {Array.<module:model/SchemasVersionResponse>} versions
 */
ServiceDetail.prototype['versions'] = undefined;

/**
 * @member {module:model/ServiceVersionDetailOrNull} active_version
 */
ServiceDetail.prototype['active_version'] = undefined;

/**
 * @member {module:model/ServiceVersionDetail} version
 */
ServiceDetail.prototype['version'] = undefined;

// Implement ServiceResponse interface:
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
_ServiceResponse["default"].prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */
_ServiceResponse["default"].prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
_ServiceResponse["default"].prototype['updated_at'] = undefined;
/**
 * A freeform descriptive note.
 * @member {String} comment
 */
_ServiceResponse["default"].prototype['comment'] = undefined;
/**
 * The name of the service.
 * @member {String} name
 */
_ServiceResponse["default"].prototype['name'] = undefined;
/**
 * Alphanumeric string identifying the customer.
 * @member {String} customer_id
 */
_ServiceResponse["default"].prototype['customer_id'] = undefined;
/**
 * The type of this service.
 * @member {module:model/ServiceResponse.TypeEnum} type
 */
_ServiceResponse["default"].prototype['type'] = undefined;
/**
 * @member {String} id
 */
_ServiceResponse["default"].prototype['id'] = undefined;
/**
 * Unused at this time.
 * @member {String} publish_key
 */
_ServiceResponse["default"].prototype['publish_key'] = undefined;
/**
 * Whether the service is paused. Services are paused due to a lack of traffic for an extended period of time. Services are resumed either when a draft version is activated or a locked version is cloned and reactivated.
 * @member {Boolean} paused
 */
_ServiceResponse["default"].prototype['paused'] = undefined;
/**
 * A list of [versions](/reference/api/services/version/) associated with the service.
 * @member {Array.<module:model/SchemasVersionResponse>} versions
 */
_ServiceResponse["default"].prototype['versions'] = undefined;
// Implement ServiceDetailAllOf interface:
/**
 * @member {module:model/ServiceVersionDetailOrNull} active_version
 */
_ServiceDetailAllOf["default"].prototype['active_version'] = undefined;
/**
 * @member {module:model/ServiceVersionDetail} version
 */
_ServiceDetailAllOf["default"].prototype['version'] = undefined;

/**
 * Allowed values for the <code>type</code> property.
 * @enum {String}
 * @readonly
 */
ServiceDetail['TypeEnum'] = {
  /**
   * value: "vcl"
   * @const
   */
  "vcl": "vcl",
  /**
   * value: "wasm"
   * @const
   */
  "wasm": "wasm"
};
var _default = ServiceDetail;
exports["default"] = _default;
