// fastly/dist/model/SettingsResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _ServiceIdAndVersion = _interopRequireDefault(require("./ServiceIdAndVersion"));
var _Settings = _interopRequireDefault(require("./Settings"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The SettingsResponse model module.
 * @module model/SettingsResponse
 * @version 4.1.1
 */
var SettingsResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SettingsResponse</code>.
   * @alias module:model/SettingsResponse
   * @implements module:model/Settings
   * @implements module:model/ServiceIdAndVersion
   */
  function SettingsResponse() {
    _classCallCheck(this, SettingsResponse);
    _Settings["default"].initialize(this);
    _ServiceIdAndVersion["default"].initialize(this);
    SettingsResponse.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(SettingsResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>SettingsResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SettingsResponse} obj Optional instance to populate.
     * @return {module:model/SettingsResponse} The populated <code>SettingsResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SettingsResponse();
        _Settings["default"].constructFromObject(data, obj);
        _ServiceIdAndVersion["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('general.default_host')) {
          obj['general.default_host'] = _ApiClient["default"].convertToType(data['general.default_host'], 'String');
        }
        if (data.hasOwnProperty('general.default_ttl')) {
          obj['general.default_ttl'] = _ApiClient["default"].convertToType(data['general.default_ttl'], 'Number');
        }
        if (data.hasOwnProperty('general.stale_if_error')) {
          obj['general.stale_if_error'] = _ApiClient["default"].convertToType(data['general.stale_if_error'], 'Boolean');
        }
        if (data.hasOwnProperty('general.stale_if_error_ttl')) {
          obj['general.stale_if_error_ttl'] = _ApiClient["default"].convertToType(data['general.stale_if_error_ttl'], 'Number');
        }
        if (data.hasOwnProperty('service_id')) {
          obj['service_id'] = _ApiClient["default"].convertToType(data['service_id'], 'String');
        }
        if (data.hasOwnProperty('version')) {
          obj['version'] = _ApiClient["default"].convertToType(data['version'], 'Number');
        }
      }
      return obj;
    }
  }]);
  return SettingsResponse;
}();
/**
 * The default host name for the version.
 * @member {String} general.default_host
 */
SettingsResponse.prototype['general.default_host'] = undefined;

/**
 * The default time-to-live (TTL) for the version.
 * @member {Number} general.default_ttl
 */
SettingsResponse.prototype['general.default_ttl'] = undefined;

/**
 * Enables serving a stale object if there is an error.
 * @member {Boolean} general.stale_if_error
 * @default false
 */
SettingsResponse.prototype['general.stale_if_error'] = false;

/**
 * The default time-to-live (TTL) for serving the stale object for the version.
 * @member {Number} general.stale_if_error_ttl
 * @default 43200
 */
SettingsResponse.prototype['general.stale_if_error_ttl'] = 43200;

/**
 * @member {String} service_id
 */
SettingsResponse.prototype['service_id'] = undefined;

/**
 * @member {Number} version
 */
SettingsResponse.prototype['version'] = undefined;

// Implement Settings interface:
/**
 * The default host name for the version.
 * @member {String} general.default_host
 */
_Settings["default"].prototype['general.default_host'] = undefined;
/**
 * The default time-to-live (TTL) for the version.
 * @member {Number} general.default_ttl
 */
_Settings["default"].prototype['general.default_ttl'] = undefined;
/**
 * Enables serving a stale object if there is an error.
 * @member {Boolean} general.stale_if_error
 * @default false
 */
_Settings["default"].prototype['general.stale_if_error'] = false;
/**
 * The default time-to-live (TTL) for serving the stale object for the version.
 * @member {Number} general.stale_if_error_ttl
 * @default 43200
 */
_Settings["default"].prototype['general.stale_if_error_ttl'] = 43200;
// Implement ServiceIdAndVersion interface:
/**
 * @member {String} service_id
 */
_ServiceIdAndVersion["default"].prototype['service_id'] = undefined;
/**
 * @member {Number} version
 */
_ServiceIdAndVersion["default"].prototype['version'] = undefined;
var _default = SettingsResponse;
exports["default"] = _default;
