// fastly/dist/model/TlsBulkCertificateResponseAttributes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Timestamps = _interopRequireDefault(require("./Timestamps"));
var _TlsBulkCertificateResponseAttributesAllOf = _interopRequireDefault(require("./TlsBulkCertificateResponseAttributesAllOf"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The TlsBulkCertificateResponseAttributes model module.
 * @module model/TlsBulkCertificateResponseAttributes
 * @version 4.1.1
 */
var TlsBulkCertificateResponseAttributes = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TlsBulkCertificateResponseAttributes</code>.
   * @alias module:model/TlsBulkCertificateResponseAttributes
   * @implements module:model/Timestamps
   * @implements module:model/TlsBulkCertificateResponseAttributesAllOf
   */
  function TlsBulkCertificateResponseAttributes() {
    _classCallCheck(this, TlsBulkCertificateResponseAttributes);
    _Timestamps["default"].initialize(this);
    _TlsBulkCertificateResponseAttributesAllOf["default"].initialize(this);
    TlsBulkCertificateResponseAttributes.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(TlsBulkCertificateResponseAttributes, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>TlsBulkCertificateResponseAttributes</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TlsBulkCertificateResponseAttributes} obj Optional instance to populate.
     * @return {module:model/TlsBulkCertificateResponseAttributes} The populated <code>TlsBulkCertificateResponseAttributes</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TlsBulkCertificateResponseAttributes();
        _Timestamps["default"].constructFromObject(data, obj);
        _TlsBulkCertificateResponseAttributesAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('deleted_at')) {
          obj['deleted_at'] = _ApiClient["default"].convertToType(data['deleted_at'], 'Date');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
        if (data.hasOwnProperty('not_after')) {
          obj['not_after'] = _ApiClient["default"].convertToType(data['not_after'], 'Date');
        }
        if (data.hasOwnProperty('not_before')) {
          obj['not_before'] = _ApiClient["default"].convertToType(data['not_before'], 'Date');
        }
        if (data.hasOwnProperty('replace')) {
          obj['replace'] = _ApiClient["default"].convertToType(data['replace'], 'Boolean');
        }
      }
      return obj;
    }
  }]);
  return TlsBulkCertificateResponseAttributes;
}();
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
TlsBulkCertificateResponseAttributes.prototype['created_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */
TlsBulkCertificateResponseAttributes.prototype['deleted_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
TlsBulkCertificateResponseAttributes.prototype['updated_at'] = undefined;

/**
 * Time-stamp (GMT) when the certificate will expire. Must be in the future to be used to terminate TLS traffic.
 * @member {Date} not_after
 */
TlsBulkCertificateResponseAttributes.prototype['not_after'] = undefined;

/**
 * Time-stamp (GMT) when the certificate will become valid. Must be in the past to be used to terminate TLS traffic.
 * @member {Date} not_before
 */
TlsBulkCertificateResponseAttributes.prototype['not_before'] = undefined;

/**
 * A recommendation from Fastly indicating the key associated with this certificate is in need of rotation.
 * @member {Boolean} replace
 */
TlsBulkCertificateResponseAttributes.prototype['replace'] = undefined;

// Implement Timestamps interface:
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
_Timestamps["default"].prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */
_Timestamps["default"].prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
_Timestamps["default"].prototype['updated_at'] = undefined;
// Implement TlsBulkCertificateResponseAttributesAllOf interface:
/**
 * Time-stamp (GMT) when the certificate will expire. Must be in the future to be used to terminate TLS traffic.
 * @member {Date} not_after
 */
_TlsBulkCertificateResponseAttributesAllOf["default"].prototype['not_after'] = undefined;
/**
 * Time-stamp (GMT) when the certificate will become valid. Must be in the past to be used to terminate TLS traffic.
 * @member {Date} not_before
 */
_TlsBulkCertificateResponseAttributesAllOf["default"].prototype['not_before'] = undefined;
/**
 * A recommendation from Fastly indicating the key associated with this certificate is in need of rotation.
 * @member {Boolean} replace
 */
_TlsBulkCertificateResponseAttributesAllOf["default"].prototype['replace'] = undefined;
var _default = TlsBulkCertificateResponseAttributes;
exports["default"] = _default;
