// fastly/dist/model/TlsCertificatesResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Pagination = _interopRequireDefault(require("./Pagination"));
var _PaginationLinks = _interopRequireDefault(require("./PaginationLinks"));
var _PaginationMeta = _interopRequireDefault(require("./PaginationMeta"));
var _TlsCertificateResponseData = _interopRequireDefault(require("./TlsCertificateResponseData"));
var _TlsCertificatesResponseAllOf = _interopRequireDefault(require("./TlsCertificatesResponseAllOf"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The TlsCertificatesResponse model module.
 * @module model/TlsCertificatesResponse
 * @version 4.1.1
 */
var TlsCertificatesResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TlsCertificatesResponse</code>.
   * @alias module:model/TlsCertificatesResponse
   * @implements module:model/Pagination
   * @implements module:model/TlsCertificatesResponseAllOf
   */
  function TlsCertificatesResponse() {
    _classCallCheck(this, TlsCertificatesResponse);
    _Pagination["default"].initialize(this);
    _TlsCertificatesResponseAllOf["default"].initialize(this);
    TlsCertificatesResponse.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(TlsCertificatesResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>TlsCertificatesResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TlsCertificatesResponse} obj Optional instance to populate.
     * @return {module:model/TlsCertificatesResponse} The populated <code>TlsCertificatesResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TlsCertificatesResponse();
        _Pagination["default"].constructFromObject(data, obj);
        _TlsCertificatesResponseAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('links')) {
          obj['links'] = _PaginationLinks["default"].constructFromObject(data['links']);
        }
        if (data.hasOwnProperty('meta')) {
          obj['meta'] = _PaginationMeta["default"].constructFromObject(data['meta']);
        }
        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_TlsCertificateResponseData["default"]]);
        }
      }
      return obj;
    }
  }]);
  return TlsCertificatesResponse;
}();
/**
 * @member {module:model/PaginationLinks} links
 */
TlsCertificatesResponse.prototype['links'] = undefined;

/**
 * @member {module:model/PaginationMeta} meta
 */
TlsCertificatesResponse.prototype['meta'] = undefined;

/**
 * @member {Array.<module:model/TlsCertificateResponseData>} data
 */
TlsCertificatesResponse.prototype['data'] = undefined;

// Implement Pagination interface:
/**
 * @member {module:model/PaginationLinks} links
 */
_Pagination["default"].prototype['links'] = undefined;
/**
 * @member {module:model/PaginationMeta} meta
 */
_Pagination["default"].prototype['meta'] = undefined;
// Implement TlsCertificatesResponseAllOf interface:
/**
 * @member {Array.<module:model/TlsCertificateResponseData>} data
 */
_TlsCertificatesResponseAllOf["default"].prototype['data'] = undefined;
var _default = TlsCertificatesResponse;
exports["default"] = _default;
