// fastly/dist/model/TlsPrivateKeyResponseAttributesAllOf.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The TlsPrivateKeyResponseAttributesAllOf model module.
 * @module model/TlsPrivateKeyResponseAttributesAllOf
 * @version 4.1.1
 */
var TlsPrivateKeyResponseAttributesAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>TlsPrivateKeyResponseAttributesAllOf</code>.
   * @alias module:model/TlsPrivateKeyResponseAttributesAllOf
   */
  function TlsPrivateKeyResponseAttributesAllOf() {
    _classCallCheck(this, TlsPrivateKeyResponseAttributesAllOf);
    TlsPrivateKeyResponseAttributesAllOf.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(TlsPrivateKeyResponseAttributesAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>TlsPrivateKeyResponseAttributesAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/TlsPrivateKeyResponseAttributesAllOf} obj Optional instance to populate.
     * @return {module:model/TlsPrivateKeyResponseAttributesAllOf} The populated <code>TlsPrivateKeyResponseAttributesAllOf</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new TlsPrivateKeyResponseAttributesAllOf();
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('key_length')) {
          obj['key_length'] = _ApiClient["default"].convertToType(data['key_length'], 'Number');
        }
        if (data.hasOwnProperty('key_type')) {
          obj['key_type'] = _ApiClient["default"].convertToType(data['key_type'], 'String');
        }
        if (data.hasOwnProperty('replace')) {
          obj['replace'] = _ApiClient["default"].convertToType(data['replace'], 'Boolean');
        }
        if (data.hasOwnProperty('public_key_sha1')) {
          obj['public_key_sha1'] = _ApiClient["default"].convertToType(data['public_key_sha1'], 'String');
        }
      }
      return obj;
    }
  }]);
  return TlsPrivateKeyResponseAttributesAllOf;
}();
/**
 * A customizable name for your private key.
 * @member {String} name
 */
TlsPrivateKeyResponseAttributesAllOf.prototype['name'] = undefined;

/**
 * The key length used to generate the private key.
 * @member {Number} key_length
 */
TlsPrivateKeyResponseAttributesAllOf.prototype['key_length'] = undefined;

/**
 * The algorithm used to generate the private key. Must be `RSA`.
 * @member {String} key_type
 */
TlsPrivateKeyResponseAttributesAllOf.prototype['key_type'] = undefined;

/**
 * A recommendation from Fastly to replace this private key and all associated certificates.
 * @member {Boolean} replace
 */
TlsPrivateKeyResponseAttributesAllOf.prototype['replace'] = undefined;

/**
 * Useful for safely identifying the key.
 * @member {String} public_key_sha1
 */
TlsPrivateKeyResponseAttributesAllOf.prototype['public_key_sha1'] = undefined;
var _default = TlsPrivateKeyResponseAttributesAllOf;
exports["default"] = _default;
