// fastly/dist/model/WafExclusionResponseDataAttributes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Timestamps = _interopRequireDefault(require("./Timestamps"));
var _WafExclusionResponseDataAttributesAllOf = _interopRequireDefault(require("./WafExclusionResponseDataAttributesAllOf"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The WafExclusionResponseDataAttributes model module.
 * @module model/WafExclusionResponseDataAttributes
 * @version 4.1.1
 */
var WafExclusionResponseDataAttributes = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>WafExclusionResponseDataAttributes</code>.
   * @alias module:model/WafExclusionResponseDataAttributes
   * @implements module:model/Timestamps
   * @implements module:model/WafExclusionResponseDataAttributesAllOf
   */
  function WafExclusionResponseDataAttributes() {
    _classCallCheck(this, WafExclusionResponseDataAttributes);
    _Timestamps["default"].initialize(this);
    _WafExclusionResponseDataAttributesAllOf["default"].initialize(this);
    WafExclusionResponseDataAttributes.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(WafExclusionResponseDataAttributes, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>WafExclusionResponseDataAttributes</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/WafExclusionResponseDataAttributes} obj Optional instance to populate.
     * @return {module:model/WafExclusionResponseDataAttributes} The populated <code>WafExclusionResponseDataAttributes</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WafExclusionResponseDataAttributes();
        _Timestamps["default"].constructFromObject(data, obj);
        _WafExclusionResponseDataAttributesAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('deleted_at')) {
          obj['deleted_at'] = _ApiClient["default"].convertToType(data['deleted_at'], 'Date');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
        if (data.hasOwnProperty('condition')) {
          obj['condition'] = _ApiClient["default"].convertToType(data['condition'], 'String');
        }
        if (data.hasOwnProperty('exclusion_type')) {
          obj['exclusion_type'] = _ApiClient["default"].convertToType(data['exclusion_type'], 'String');
        }
        if (data.hasOwnProperty('logging')) {
          obj['logging'] = _ApiClient["default"].convertToType(data['logging'], 'Boolean');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('number')) {
          obj['number'] = _ApiClient["default"].convertToType(data['number'], 'Number');
        }
        if (data.hasOwnProperty('variable')) {
          obj['variable'] = _ApiClient["default"].convertToType(data['variable'], 'String');
        }
      }
      return obj;
    }
  }]);
  return WafExclusionResponseDataAttributes;
}();
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
WafExclusionResponseDataAttributes.prototype['created_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */
WafExclusionResponseDataAttributes.prototype['deleted_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
WafExclusionResponseDataAttributes.prototype['updated_at'] = undefined;

/**
 * A conditional expression in VCL used to determine if the condition is met.
 * @member {String} condition
 */
WafExclusionResponseDataAttributes.prototype['condition'] = undefined;

/**
 * The type of exclusion.
 * @member {module:model/WafExclusionResponseDataAttributes.ExclusionTypeEnum} exclusion_type
 */
WafExclusionResponseDataAttributes.prototype['exclusion_type'] = undefined;

/**
 * Whether to generate a log upon matching.
 * @member {Boolean} logging
 * @default true
 */
WafExclusionResponseDataAttributes.prototype['logging'] = true;

/**
 * Name of the exclusion.
 * @member {String} name
 */
WafExclusionResponseDataAttributes.prototype['name'] = undefined;

/**
 * A numeric ID identifying a WAF exclusion.
 * @member {Number} number
 */
WafExclusionResponseDataAttributes.prototype['number'] = undefined;

/**
 * The variable to exclude. An optional selector can be specified after the variable separated by a colon (`:`) to restrict the variable to a particular parameter. Required for `exclusion_type=variable`.
 * @member {module:model/WafExclusionResponseDataAttributes.VariableEnum} variable
 */
WafExclusionResponseDataAttributes.prototype['variable'] = undefined;

// Implement Timestamps interface:
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
_Timestamps["default"].prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */
_Timestamps["default"].prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
_Timestamps["default"].prototype['updated_at'] = undefined;
// Implement WafExclusionResponseDataAttributesAllOf interface:
/**
 * A conditional expression in VCL used to determine if the condition is met.
 * @member {String} condition
 */
_WafExclusionResponseDataAttributesAllOf["default"].prototype['condition'] = undefined;
/**
 * The type of exclusion.
 * @member {module:model/WafExclusionResponseDataAttributesAllOf.ExclusionTypeEnum} exclusion_type
 */
_WafExclusionResponseDataAttributesAllOf["default"].prototype['exclusion_type'] = undefined;
/**
 * Whether to generate a log upon matching.
 * @member {Boolean} logging
 * @default true
 */
_WafExclusionResponseDataAttributesAllOf["default"].prototype['logging'] = true;
/**
 * Name of the exclusion.
 * @member {String} name
 */
_WafExclusionResponseDataAttributesAllOf["default"].prototype['name'] = undefined;
/**
 * A numeric ID identifying a WAF exclusion.
 * @member {Number} number
 */
_WafExclusionResponseDataAttributesAllOf["default"].prototype['number'] = undefined;
/**
 * The variable to exclude. An optional selector can be specified after the variable separated by a colon (`:`) to restrict the variable to a particular parameter. Required for `exclusion_type=variable`.
 * @member {module:model/WafExclusionResponseDataAttributesAllOf.VariableEnum} variable
 */
_WafExclusionResponseDataAttributesAllOf["default"].prototype['variable'] = undefined;

/**
 * Allowed values for the <code>exclusion_type</code> property.
 * @enum {String}
 * @readonly
 */
WafExclusionResponseDataAttributes['ExclusionTypeEnum'] = {
  /**
   * value: "rule"
   * @const
   */
  "rule": "rule",
  /**
   * value: "variable"
   * @const
   */
  "variable": "variable",
  /**
   * value: "waf"
   * @const
   */
  "waf": "waf"
};

/**
 * Allowed values for the <code>variable</code> property.
 * @enum {String}
 * @readonly
 */
WafExclusionResponseDataAttributes['VariableEnum'] = {
  /**
   * value: "req.cookies"
   * @const
   */
  "req.cookies": "req.cookies",
  /**
   * value: "req.headers"
   * @const
   */
  "req.headers": "req.headers",
  /**
   * value: "req.post"
   * @const
   */
  "req.post": "req.post",
  /**
   * value: "req.post_filename"
   * @const
   */
  "req.post_filename": "req.post_filename",
  /**
   * value: "req.qs"
   * @const
   */
  "req.qs": "req.qs",
  /**
   * value: "null"
   * @const
   */
  "null": "null"
};
var _default = WafExclusionResponseDataAttributes;
exports["default"] = _default;
