// fastly/dist/model/WafFirewallVersionResponseDataAttributes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _Timestamps = _interopRequireDefault(require("./Timestamps"));
var _WafFirewallVersionResponseDataAttributesAllOf = _interopRequireDefault(require("./WafFirewallVersionResponseDataAttributesAllOf"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); } /**
                                                                                                                                                                                                                                                                                                                                                                                               * Fastly API
                                                                                                                                                                                                                                                                                                                                                                                               * Via the Fastly API you can perform any of the operations that are possible within the management console,  including creating services, domains, and backends, configuring rules or uploading your own application code, as well as account operations such as user administration and billing reports. The API is organized into collections of endpoints that allow manipulation of objects related to Fastly services and accounts. For the most accurate and up-to-date API reference content, visit our [Developer Hub](https://developer.fastly.com/reference/api/) 
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * The version of the OpenAPI document: 1.0.0
                                                                                                                                                                                                                                                                                                                                                                                               * Contact: oss@fastly.com
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               * NOTE: This class is auto generated.
                                                                                                                                                                                                                                                                                                                                                                                               * Do not edit the class manually.
                                                                                                                                                                                                                                                                                                                                                                                               *
                                                                                                                                                                                                                                                                                                                                                                                               */
/**
 * The WafFirewallVersionResponseDataAttributes model module.
 * @module model/WafFirewallVersionResponseDataAttributes
 * @version 4.1.1
 */
var WafFirewallVersionResponseDataAttributes = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>WafFirewallVersionResponseDataAttributes</code>.
   * @alias module:model/WafFirewallVersionResponseDataAttributes
   * @implements module:model/Timestamps
   * @implements module:model/WafFirewallVersionResponseDataAttributesAllOf
   */
  function WafFirewallVersionResponseDataAttributes() {
    _classCallCheck(this, WafFirewallVersionResponseDataAttributes);
    _Timestamps["default"].initialize(this);
    _WafFirewallVersionResponseDataAttributesAllOf["default"].initialize(this);
    WafFirewallVersionResponseDataAttributes.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(WafFirewallVersionResponseDataAttributes, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>WafFirewallVersionResponseDataAttributes</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/WafFirewallVersionResponseDataAttributes} obj Optional instance to populate.
     * @return {module:model/WafFirewallVersionResponseDataAttributes} The populated <code>WafFirewallVersionResponseDataAttributes</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new WafFirewallVersionResponseDataAttributes();
        _Timestamps["default"].constructFromObject(data, obj);
        _WafFirewallVersionResponseDataAttributesAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }
        if (data.hasOwnProperty('deleted_at')) {
          obj['deleted_at'] = _ApiClient["default"].convertToType(data['deleted_at'], 'Date');
        }
        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }
        if (data.hasOwnProperty('active')) {
          obj['active'] = _ApiClient["default"].convertToType(data['active'], 'Boolean');
        }
        if (data.hasOwnProperty('active_rules_fastly_block_count')) {
          obj['active_rules_fastly_block_count'] = _ApiClient["default"].convertToType(data['active_rules_fastly_block_count'], 'Number');
        }
        if (data.hasOwnProperty('active_rules_fastly_log_count')) {
          obj['active_rules_fastly_log_count'] = _ApiClient["default"].convertToType(data['active_rules_fastly_log_count'], 'Number');
        }
        if (data.hasOwnProperty('active_rules_fastly_score_count')) {
          obj['active_rules_fastly_score_count'] = _ApiClient["default"].convertToType(data['active_rules_fastly_score_count'], 'Number');
        }
        if (data.hasOwnProperty('active_rules_owasp_block_count')) {
          obj['active_rules_owasp_block_count'] = _ApiClient["default"].convertToType(data['active_rules_owasp_block_count'], 'Number');
        }
        if (data.hasOwnProperty('active_rules_owasp_log_count')) {
          obj['active_rules_owasp_log_count'] = _ApiClient["default"].convertToType(data['active_rules_owasp_log_count'], 'Number');
        }
        if (data.hasOwnProperty('active_rules_owasp_score_count')) {
          obj['active_rules_owasp_score_count'] = _ApiClient["default"].convertToType(data['active_rules_owasp_score_count'], 'Number');
        }
        if (data.hasOwnProperty('active_rules_trustwave_block_count')) {
          obj['active_rules_trustwave_block_count'] = _ApiClient["default"].convertToType(data['active_rules_trustwave_block_count'], 'Number');
        }
        if (data.hasOwnProperty('active_rules_trustwave_log_count')) {
          obj['active_rules_trustwave_log_count'] = _ApiClient["default"].convertToType(data['active_rules_trustwave_log_count'], 'Number');
        }
        if (data.hasOwnProperty('last_deployment_status')) {
          obj['last_deployment_status'] = _ApiClient["default"].convertToType(data['last_deployment_status'], 'String');
        }
        if (data.hasOwnProperty('deployed_at')) {
          obj['deployed_at'] = _ApiClient["default"].convertToType(data['deployed_at'], 'String');
        }
        if (data.hasOwnProperty('error')) {
          obj['error'] = _ApiClient["default"].convertToType(data['error'], 'String');
        }
      }
      return obj;
    }
  }]);
  return WafFirewallVersionResponseDataAttributes;
}();
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
WafFirewallVersionResponseDataAttributes.prototype['created_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */
WafFirewallVersionResponseDataAttributes.prototype['deleted_at'] = undefined;

/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
WafFirewallVersionResponseDataAttributes.prototype['updated_at'] = undefined;

/**
 * Whether a specific firewall version is currently deployed.
 * @member {Boolean} active
 */
WafFirewallVersionResponseDataAttributes.prototype['active'] = undefined;

/**
 * The number of active Fastly rules set to block.
 * @member {Number} active_rules_fastly_block_count
 */
WafFirewallVersionResponseDataAttributes.prototype['active_rules_fastly_block_count'] = undefined;

/**
 * The number of active Fastly rules set to log.
 * @member {Number} active_rules_fastly_log_count
 */
WafFirewallVersionResponseDataAttributes.prototype['active_rules_fastly_log_count'] = undefined;

/**
 * The number of active Fastly rules set to score.
 * @member {Number} active_rules_fastly_score_count
 */
WafFirewallVersionResponseDataAttributes.prototype['active_rules_fastly_score_count'] = undefined;

/**
 * The number of active OWASP rules set to block.
 * @member {Number} active_rules_owasp_block_count
 */
WafFirewallVersionResponseDataAttributes.prototype['active_rules_owasp_block_count'] = undefined;

/**
 * The number of active OWASP rules set to log.
 * @member {Number} active_rules_owasp_log_count
 */
WafFirewallVersionResponseDataAttributes.prototype['active_rules_owasp_log_count'] = undefined;

/**
 * The number of active OWASP rules set to score.
 * @member {Number} active_rules_owasp_score_count
 */
WafFirewallVersionResponseDataAttributes.prototype['active_rules_owasp_score_count'] = undefined;

/**
 * The number of active Trustwave rules set to block.
 * @member {Number} active_rules_trustwave_block_count
 */
WafFirewallVersionResponseDataAttributes.prototype['active_rules_trustwave_block_count'] = undefined;

/**
 * The number of active Trustwave rules set to log.
 * @member {Number} active_rules_trustwave_log_count
 */
WafFirewallVersionResponseDataAttributes.prototype['active_rules_trustwave_log_count'] = undefined;

/**
 * The status of the last deployment of this firewall version.
 * @member {module:model/WafFirewallVersionResponseDataAttributes.LastDeploymentStatusEnum} last_deployment_status
 */
WafFirewallVersionResponseDataAttributes.prototype['last_deployment_status'] = undefined;

/**
 * Time-stamp (GMT) indicating when the firewall version was last deployed.
 * @member {String} deployed_at
 */
WafFirewallVersionResponseDataAttributes.prototype['deployed_at'] = undefined;

/**
 * Contains error message if the firewall version fails to deploy.
 * @member {String} error
 */
WafFirewallVersionResponseDataAttributes.prototype['error'] = undefined;

// Implement Timestamps interface:
/**
 * Date and time in ISO 8601 format.
 * @member {Date} created_at
 */
_Timestamps["default"].prototype['created_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} deleted_at
 */
_Timestamps["default"].prototype['deleted_at'] = undefined;
/**
 * Date and time in ISO 8601 format.
 * @member {Date} updated_at
 */
_Timestamps["default"].prototype['updated_at'] = undefined;
// Implement WafFirewallVersionResponseDataAttributesAllOf interface:
/**
 * Whether a specific firewall version is currently deployed.
 * @member {Boolean} active
 */
_WafFirewallVersionResponseDataAttributesAllOf["default"].prototype['active'] = undefined;
/**
 * The number of active Fastly rules set to block.
 * @member {Number} active_rules_fastly_block_count
 */
_WafFirewallVersionResponseDataAttributesAllOf["default"].prototype['active_rules_fastly_block_count'] = undefined;
/**
 * The number of active Fastly rules set to log.
 * @member {Number} active_rules_fastly_log_count
 */
_WafFirewallVersionResponseDataAttributesAllOf["default"].prototype['active_rules_fastly_log_count'] = undefined;
/**
 * The number of active Fastly rules set to score.
 * @member {Number} active_rules_fastly_score_count
 */
_WafFirewallVersionResponseDataAttributesAllOf["default"].prototype['active_rules_fastly_score_count'] = undefined;
/**
 * The number of active OWASP rules set to block.
 * @member {Number} active_rules_owasp_block_count
 */
_WafFirewallVersionResponseDataAttributesAllOf["default"].prototype['active_rules_owasp_block_count'] = undefined;
/**
 * The number of active OWASP rules set to log.
 * @member {Number} active_rules_owasp_log_count
 */
_WafFirewallVersionResponseDataAttributesAllOf["default"].prototype['active_rules_owasp_log_count'] = undefined;
/**
 * The number of active OWASP rules set to score.
 * @member {Number} active_rules_owasp_score_count
 */
_WafFirewallVersionResponseDataAttributesAllOf["default"].prototype['active_rules_owasp_score_count'] = undefined;
/**
 * The number of active Trustwave rules set to block.
 * @member {Number} active_rules_trustwave_block_count
 */
_WafFirewallVersionResponseDataAttributesAllOf["default"].prototype['active_rules_trustwave_block_count'] = undefined;
/**
 * The number of active Trustwave rules set to log.
 * @member {Number} active_rules_trustwave_log_count
 */
_WafFirewallVersionResponseDataAttributesAllOf["default"].prototype['active_rules_trustwave_log_count'] = undefined;
/**
 * The status of the last deployment of this firewall version.
 * @member {module:model/WafFirewallVersionResponseDataAttributesAllOf.LastDeploymentStatusEnum} last_deployment_status
 */
_WafFirewallVersionResponseDataAttributesAllOf["default"].prototype['last_deployment_status'] = undefined;
/**
 * Time-stamp (GMT) indicating when the firewall version was last deployed.
 * @member {String} deployed_at
 */
_WafFirewallVersionResponseDataAttributesAllOf["default"].prototype['deployed_at'] = undefined;
/**
 * Contains error message if the firewall version fails to deploy.
 * @member {String} error
 */
_WafFirewallVersionResponseDataAttributesAllOf["default"].prototype['error'] = undefined;

/**
 * Allowed values for the <code>last_deployment_status</code> property.
 * @enum {String}
 * @readonly
 */
WafFirewallVersionResponseDataAttributes['LastDeploymentStatusEnum'] = {
  /**
   * value: "null"
   * @const
   */
  "null": "null",
  /**
   * value: "pending"
   * @const
   */
  "pending": "pending",
  /**
   * value: "in progress"
   * @const
   */
  "in progress": "in progress",
  /**
   * value: "completed"
   * @const
   */
  "completed": "completed",
  /**
   * value: "failed"
   * @const
   */
  "failed": "failed"
};
var _default = WafFirewallVersionResponseDataAttributes;
exports["default"] = _default;
